lib/db.js

```diff
@@ -215,6 +215,23 @@ module.exports = function (
 
   // READ
 
+  DB.prototype.checkPassword = function (uid, verifyHash) {
+    verifyHash = Buffer(verifyHash).toString('hex')
+    log.trace({ op: 'DB.checkPassword', uid: uid, verifyHash: verifyHash })
+    return this.pool.post('/account/' + uid.toString('hex') + '/checkPassword',
+      {
+        'verifyHash': verifyHash
+      })
+      .then(
+        function (body) {
+          return uid.toString('hex') === body.uid
+        },
+        function (err) {
+          return false
+        }
+      )
```

Comment:
Alternatively, you could do something like the `.accountExists()` does below. The resolved function just returns true, the rejected function checks for a 404 and returns false, else throws any other (bad) error. Not sure if that gets us anything extra but it does mirror what `accountExists` does.